### PR TITLE
[cartoonnetwork] Fixed AttributeError: 'NoneType' object has no attribute 'groups'

### PR DESCRIPTION
**Pull request by TheDaChicken**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Fixed on Cartoonnetwork.py "AttributeError: 'NoneType' object has no attribute 'groups" by changing cvp to ngtv, using #16930  and by getting a id a different way since the way that was in #16930 wasn't working. There is probably a better way of getting the id but it works and I have tested it a few times.

### CODE REVIEW

**Comment by TheDaChicken** [comment]
Do I need to ping anyone? This has been open and no one has checked this one out.

**Comment by TheDaChicken** [comment]
I will get to that! Sorry about that.

**Comment by TheDaChicken** [comment]
Ok! You should check out the code now. You can make any changes if you want!

**Comment by TheDaChicken** [comment]
@JamesTheAwesomeDude Can you try out the latest change? I changed some things and I wanna see if it still works without asking "Please sign in with your tv provider".

**Comment by tgray** [comment]
Was this `'NoneType' object has no attribute 'groups'` error ever fixed?

**Comment by TheDaChicken** [comment]
This is fixed in this pull request but never ended up in the master.

**Comment by TheDaChicken** [comment]
@dstftw I wasn't going to ping you but whatever. Can you check out my code? I think its up to your standards. At Least think about adding the code to the master.

**Comment by dstftw** [comment]
1. I've already pointed out: read code conventions and fix your code.
2. You must provide tv provider account credentials for testing.

**Comment by TheDaChicken** [comment]
> * I've already pointed out: read code conventions and fix your code.
Well, Ya. I got that.
> * You must provide tv provider account credentials for testing.
You don't need a provider account for testing. There are free episodes available. Some weekends, all episodes are free. Like This: https://www.cartoonnetwork.com/video/gumball/the-console-episode.html (Full Unlocked Episode) You may not even be able to go to the link since you added "geo-restricted" to this.

**Comment by dstftw** [comment]
There is an **auth related code** in the extractor to test which tv providee account is required.

**Comment by TheDaChicken** [comment]
> There is an **auth related code** in the extractor to test which tv providee account is required.
Well, Since I can't give you my account (Mostly because passwords, and it not even supported by youtube-dl and came from a pull request that wasn't accept)

**Comment by JamesTheAwesomeDude** [comment]
@dstftw: Am I correct in understanding that this pull request — which **fixes** #17224,  a **Blocker** for the _entire `CartoonNetwork`handler_ — is not going to be accepted

because it doesn't fix a different, _tangentially related_ bug, that nobody's reported, only affects a subset of users, and is furthermore _completely eclipsed by_ the very bug this PR fixes? 

Or is it something deeper like "the auth code was already broken, but this PR breaks it even further, and will lead to a bigger mess to clean up in the long run"?

-----

**tl;dr: What's wrong with this PR?**

Did @TheDaChicken (in the process of fixing this Blocker) make some changes to the auth code that were less conservative than they could/should have been?

Was #17224 part of a cascading failure of the Turner code generally, and this PR only patches a symptom without solving the problem?

Did he just indent his code wrong?

**What do we gotta do to make this patch acceptable so #17224 can be fixed?**

**Comment by TheDaChicken** [comment]
Well, It seems some other version was added in the master. Even tho, I think I first had cleaner code. I don't get this anymore.

**Comment by remitamine** [comment]
> He pull request was made after mine. :/ and accepted immediately but his code was up to expectations and I fixed my code up to expectations long before it.

not it wasn't up to expectations, you didn't address what @dstftw asked you to fix:
> 1. I've already pointed out: read code conventions and fix your code.
- a lot of duplication.
- `_html_search_regex` is not used properly.
- `season_id` should be a string, not an integer, and should not be filled with `seriesId`.
- the code does not extract any fallback especially for mandatory fields(id and title).
- the regular expressions are too relaxed.
- ...
while in the code i added, i have fixed most of the coding problems and checked that the auth related code is working.
> This is annoying.

what is annoying me is when @JamesTheAwesomeDude come and suggest that we're blocking fixes from being merged.
> I think they was choosing from preference of person.

you don't understand how and why the code has been added and you're just accusing us of favoring some people over others.

there is something that a lot of people coming here are not getting, we are volunteers(we're not employees and we don't get paid for the work we do), we don't owe anything to any user, we work on the (issues/pull requests) that interest us in our free time, we have our private life and things that we to take care of(study, work...), we don't have to fix every issue and add every requested feature and reply to every comment, it's actually almost impossible with the large amount of issues and pull requests and the small team.

**Comment by TheDaChicken** [comment]
> * `_html_search_regex` is not used properly.
I know it sounds stupid, but... Got it! I didn't know. 



> * `season_id` should be a string, not an integer, and should not be filled with `seriesId`.
Oh! Sorry, that is a mistake.



> you don't understand how and why the code has been added and you're just accusing us of favoring some people over others.
> 
> there is something that a lot of people coming here are not getting, we are volunteers(we're not employees and we don't get paid for the work we do), we don't owe anything to any user, we work on the (issues/pull requests) that interest us in our free time, we have our private life and things that we to take care of(study, work...), we don't have to fix every issue and add every requested feature and reply to every comment, it's actually almost impossible with the large amount of issues and pull requests and the small team.

Well, Well, What do you do when you don't know what your doing wrong? Yes, I understand. :/ I was being rude. I was just angry.
